grpc-spring-boot-starter/src/main/java/org/lognet/springboot/grpc/GRpcServerRunner.java

```diff
@@ -147,6 +152,19 @@ public void destroy() throws Exception {
             log.info("Shutting down gRPC server ...");
             s.getServices().forEach(def->healthStatusManager.clearStatus(def.getServiceDescriptor().getName()));
             s.shutdown();
+            int shutdownGrace = gRpcServerProperties.getShutdownGrace();
+            try {
+                // If shutdownGrace is 0, then don't call awaitTermination
+                if (shutdownGrace < 0) {
+                    s.awaitTermination();
+                } else if (shutdownGrace > 0) {
+                    s.awaitTermination(shutdownGrace, TimeUnit.SECONDS);
```

Comment:
@dounan @jvmlet  how do you think, probably here you can use something like this:

```java
if (!s.awaitTermination(shutdownGrace, TimeUnit.SECONDS)) {
    s.shutdownNow();
    s.awaitTermination(shutdownGrace, TimeUnit.SECONDS);
}

```
The purpose of this construction is to release the socket before the first termination (`s.shutdown();`), but to shutdown gracefully the whole server.

Reply:
Thanks for your suggestion @mchernyakov! That's an interesting thought.

I'm not sure what the additional call to `shutdownNow` will get us. I think the client behavior will basically be the same, they'll see error code 14 with "socket closed". If there is a good reason to add it, I can do it.

The second `awaitTermination` is likely not needed since `shutdownNow` should cancel all preexisting calls.

Reply:
@dounan it depends on our goals. If we wanted to cut the connections and interrupt current executions I think we should use `s.shutdownNow()`, otherwise just `s.shutdown()`.

As far as I understand you want to keep them running till the end. But it some cases it is good to interrupt executions because they could run forever.